### DISCUSSION TITLE
[FEATURE REQUEST] salt.states.virt: support on_crash

### DISCUSSION THREAD

**Discussion by dseomn** [saltstack/salt#67632] [Feature Requests]
**Describe the solution you'd like**
It would be nice if salt.states.virt.running and/or salt.states.virt.defined supported managing the `on_crash` setting, or possibly all of the [events settings](https://libvirt.org/formatdomain.html#events-configuration).

**Comment by garethgreenaway** [comment]
@cbosdo FYI.

**Comment by cbosdo** [comment]
Thanks for the feature request. The `on_reboot` event is already handled. The `on_crash`, `on_poweroff` and `on_lockfailure` could be implemented in a similar way.

For anyone who would like to have a go at implementing those, here is the commit adding the `on_reboot` parameter to get inspiration from: https://github.com/saltstack/salt/commit/9714bd00fac8080d0f294532b98a49542c70051f. I would be happy to help someone submit a pull request for this.

**Comment by darshanip** [comment]
I'll pick it up, will be raising PR sortly. @cbosdo
